Dalamud/Game/Gui/ContextMenu/CharacterData.cs

```diff
@@ -0,0 +1,472 @@
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Dalamud.Memory;
+using FFXIVClientStructs.FFXIV.Client.UI.Info;
+
+namespace Dalamud.Game.Gui.ContextMenu;
+
+/// <summary>
+/// Dalamud wrapper around a ClientStructs CharacterData.
+/// </summary>
+[StructLayout(LayoutKind.Explicit, Size = StructSizeInBytes)]
+public unsafe struct CharacterData
```

Comment:
Is there (a) a reason this isn’t a Class, and (b) a reason why this has to be inside ContextMenu? Most other wrappers are classes to my knowledge.

Reply:
I’m unsure GameInventoryItem was correct with their decision though they also use it for certain equality magic. If there’s no compelling reason to make this a struct, however, I think it would be best to convert to a class. 

I’d also suggest moving it somewhere more appropriate as this is for InfoProxy in general and I can see this being useful elsewhere later. This would also likely benefit a rename to CharacterSearchEntry or something similar.

Reply:
***At the time*** GameInventoryItem needed to be a struct.

It was created with the intent of being able to wrap an address (like how Dalamud GameObject wraps an address), and to be a way to hold a copy of the data if needed.

**At the time** this was required because we were scanning the inventory memory every frame, and it was definitely a time-sensitive task. Initial version of the scanner were very slow.

Immediately prior to releasing the GameInventory service it was discovered that there's a inventory changed flag in one of the game systems, we now use this flag to determine if the inventory should be scanned.

**Currently** there's not really much need for most of the optimizations in GameInventoryItem, it could be cut down significantly and reduced back down to a standard class object wrapper. *Ideally* I'd love to just return the struct from ClientStructs itself instead of using a wrapper, but that's not practical at this time.